onedal/_device_offload.py

```diff
@@ -178,11 +178,24 @@ def _convert_to_dpnp(array):
 
 
 def support_usm_ndarray(freefunc=False, queue_param=True):
+    """
+    Handles USMArray input. Puts SYCLQueue from data to decorated function arguments.
+    Converts output of decorated function to dpnp.ndarray if input was of this type.
+
+    Parameters
+    ----------
+    freefunc (bool) : Set to True if decorates free function.
+    queue_param (bool) : Set to True if queue from data should be provided to the decorated function args.
+                         Queue will not be changed if provided explicitly.
+    """
+
     def decorator(func):
         def wrapper_impl(obj, *args, **kwargs):
             usm_iface = _extract_usm_iface(*args, **kwargs)
             data_queue, hostargs, hostkwargs = _get_host_inputs(*args, **kwargs)
-            if queue_param:
+            if queue_param and not (
+                "queue" in hostkwargs and hostkwargs["queue"] is not None
+            ):
```

Comment:
```suggestion
            if not ("queue" in hostkwargs and hostkwargs["queue"] is not None):
```
This param were added on #1832, now it looks overkill for this decorator. Maybe it make sense just remove `queue_param` usage and param from `support_usm_ndarray`and just leave this condition checking.

Just needed to remove all `queue_param=False` from everywhere.

Reply:
Outdated.